spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java

```diff
@@ -543,11 +539,7 @@ public void setGatewayAddress(final String gatewayAddress) {
     @Deprecated
     @DeprecatedConfigurationProperty(replacement = "camunda.client.zeebe.grpc-address")
     public URI getGrpcAddress() {
-      if (grpcAddress != null) {
-        return grpcAddress;
-      } else {
-        return DEFAULT.getGrpcAddress();
-      }
+      return grpcAddress;
```

Comment:
"getGrpcAddress" and "getRestAddress":
Return null if not set so default value could be set after checking new
config. This done because we check the old config first and only then -
the new one.